lib/pbench/server/api/__init__.py

```diff
@@ -286,7 +286,27 @@ def post(self):
             app.logger.debug("Bad file extension received")
             abort(400, message="File extension not supported. Only .xz")
 
+        try:
+            content_length = int(request.headers.get("Content-Length"))
+        except ValueError:
+            abort(400, message="Invalid content-length header, not an integer")
+        except Exception:
+            abort(400, message="Missing required content-length header")
+        else:
+            if content_length > _MAX_UPLOAD_DATA_SIZE:
+                abort(
+                    400,
+                    message="Content object too large, keep it at 1 GB"
+                    f" ({content_length:d}) and  under",
```

Comment:
Extra space; you might want to use the [`humanize`](https://pypi.org/project/humanize/) module to avoid embedding the constant of `1 GB` in the abort message string.

Reply:
Small concern: We will have to add this module in the requirement file, is it fine to add one more dependency just for this line? (unless we plan to use the module in other part of the code)